internal/users/idgenerator.go

```diff
@@ -270,37 +151,44 @@ func getIDCandidate(minID, maxID uint32, usedIDs []uint32) (id uint32, uniqueIDs
 	}
 
 	// Try IDs starting from the preferred ID up to the maximum ID.
-	for id := preferredID; ; id++ {
-		// Sanitize the ID to ensure that it's not in a restricted range.
-		id = adjustIDForSafeRanges(id)
-
-		if id > maxID {
-			break
+	// Overflows are avoided by the "id < math.MaxUint32" condition.
+	for id := preferredID; id <= maxID && id < math.MaxUint32; id++ {
+		if isReservedID(id) {
+			continue
 		}
 
-		if pos, found := slices.BinarySearch(usedIDs, id); !found {
-			return id, pos, nil
+		pos, found := slices.BinarySearch(usedIDs, id)
+		if found {
+			continue
 		}
 
-		if id == math.MaxUint32 {
-			break // Avoid overflow
-		}
+		return id, pos, nil
 	}
 
 	// Fallback: try IDs from the minimum ID up to the preferred ID.
+	// Overflows are avoided by "id < preferredID" condition, because
+	// preferredID is a uint32, so the condition must be false when
+	// id == math.MaxUint32).
 	for id := minID; id < preferredID && id <= maxID; id++ {
-		if pos, found := slices.BinarySearch(usedIDs, id); !found {
-			return id, pos, nil
+		if isReservedID(id) {
+			continue
+		}
+
+		pos, found := slices.BinarySearch(usedIDs, id)
+		if found {
+			continue
 		}
 
-		// Overflows are avoided by the loop condition (id < preferredID, where
-		// preferredID is a uint32, so the condition must be false when
-		// id == math.MaxUint32).
+		return id, pos, nil
 	}
 
 	return 0, -1, errors.New("no available ID in range")
 }
 
+func isReservedID(id uint32) bool {
```

Comment:
Should we also add a comment similar to previous?

```
// isReservedID checks if the ID is a value is not a linux system reserved value.
// See: https://systemd.io/UIDS-GIDS/
```

Reply:
I had this comment on the constants instead:
```
// Special Linux UIDs, see https://systemd.io/UIDS-GIDS/
const (
	rootID         uint32 = 0
	nobodyID       uint32 = 65534
	uidT32MinusOne uint32 = math.MaxUint32
	uidT16MinusOne uint32 = math.MaxUint16
)

```